packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -148,6 +148,15 @@ export abstract class HandlebarsNodeVisitors extends Parser {
         );
 
       case TokenizerState.beforeAttributeName:
+        if (this.currentNode && this.currentNode.type === 'EndTag') {
```

Comment:
I'm surprised we enter this state during a close tag 🤔

Reply:
I was surprised as well. This state is hit in this scenario: `<div></div {{some-modifier}}>`, and it was actually already throwing this error`"Error: expected start tag"` which is triggered in the `this.currentStartTag` call on line 160, but doesn't really describe the error accurately. 

`currentStartTag` snippet:
```js
  get currentStartTag(): Tag<'StartTag'> {
    let node = this.currentNode;
    assert(node && node.type === 'StartTag', 'expected start tag');
    return node as Tag<'StartTag'>;
  }
```

If you comment out this `if` check, two of the tests I added will fail, throwing the `expected start tag` error instead of the one added in this PR.

Reply:
OK, gotcha, thank you for explaining.

I _believe_ what is going on is that the HTML spec requires the tagname **immediately** after the `</`.

See the spec info here:

https://html.spec.whatwg.org/multipage/parsing.html#end-tag-open-state

And our implementation here:

https://github.com/tildeio/simple-html-tokenizer/blob/691662c9f00aec719d0eec8e622ed99935119c55/src/evented-tokenizer.ts#L482-L491

I think _essentially_ we want an `else` to the `if` above that throws a useful error (RE: invalid end tag). I think that would look something like:

```js
    endTagOpen() {
      let char = this.consume();

      if (char === '@' || char === ':' || isAlpha(char)) {
        this.transitionTo(TokenizerState.endTagName);
        this.tagNameBuffer = '';
        this.delegate.beginEndTag();
        this.appendToTagName(char);
      } else {
        this.delegate.reportSyntaxError('invalid closing tag');
      }
    }
```

Would you be interested in working on a fix over in simple-html-tokenizer also?

Reply:
I also think this is a bit wrong:

https://github.com/tildeio/simple-html-tokenizer/blob/691662c9f00aec719d0eec8e622ed99935119c55/src/evented-tokenizer.ts#L266-L282

We have created the `endTagName` state, and made it _essentially_ the same as the `tagName` state (the spec actually says to use the `tagName` state ([see here](https://html.spec.whatwg.org/multipage/parsing.html#end-tag-open-state)), but thats a tad bit difficult for us). But honestly, I don't think we actually are allowed to enter those other states (`beforeAttributeName` and `selfClosingStartTag`). We should probably add errors there too.

Reply:
I'll make an attempt at fixing those issues in Simple HTML Tokenizer this weekend. I'll open a PR with the fixes, or an issue, if I'm unable to come up with solutions

Thanks for looking into this!

Reply:
Made a PR here https://github.com/tildeio/simple-html-tokenizer/pull/73 would love feedback 😄